packages/vsx-registry/src/browser/vsx-extensions-contribution.ts

```diff
@@ -107,4 +172,22 @@ export class VSXExtensionsContribution extends AbstractViewContribution<VSXExten
         }
         return false;
     }
+
+    /**
+     * Installs a local .vsix file after prompting the `Open File` dialog. Resolves to the URI of the file.
+     */
+    protected async installFromVSIX(): Promise<URI | undefined> {
+        const props: OpenFileDialogProps = {
+            title: VSXExtensionsCommands.INSTALL_FROM_VSIX.dialogLabel,
+            openLabel: 'Install',
+            filters: { 'VSIX Extensions (*.vsix)': ['vsix'] }
+        };
+        const [rootStat] = await this.workspaceService.roots;
+        const destinationFileUri = await this.fileDialogService.showOpenDialog(props, rootStat);
```

Comment:
We can probably omit the `rootStat` (we wouldn't need to start the lookup at the workspace root).